tools/deployment-cli-tools/ch_cli_tools/codefresh.py

```diff
@@ -33,6 +34,27 @@ def literal_presenter(dumper, data):
 
 yaml.add_representer(str, literal_presenter)
 
+def get_main_domain(url):
+    try:
+        url = url.split("//")[1].split("/")[0]
+        if "gitlab" in url:
+            return "gitlab"
+        if "bitbucket" in url:
+            return "bitbucket"
+        return "github"
+    except:
+        return "${{ DEFAULT_REPO }}"
+
+def clone_step_spec(conf: GitDependencyConfig, context_path: str):
+    return {
```

Comment:
@filippomc question: can we also implement support for cloning private git repos?
I know for git we can do it using an access token but I prefer that we don't store the access token in the "deployment" git repo

Reply:
Private repos should be supported in codefresh as it uses your login, While the local machine should work with git:// paths assuming that the ssh keys are set. Token based authentication looks potentially useful but my preference would be to put that support on a separate issue, unless there is already a practical use case